From ecb903e48822bd90650bdd64fe80754e3e9664cb Mon Sep 17 00:00:00 2001
From: Bastian Beischer <bastian.beischer@gmail.com>
Date: Fri, 2 Sep 2016 13:05:18 +0200
Subject: [PATCH] Fix display of user avatars. (#684)

QFile::exists("...") does not understand file:// URLs, at least in Qt
5.7.0 and Qt 4.8.7.
---
 src/greeter/UserModel.cpp | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/greeter/UserModel.cpp b/src/greeter/UserModel.cpp
index 41a9f10..94c492d 100644
--- a/src/greeter/UserModel.cpp
+++ b/src/greeter/UserModel.cpp
@@ -107,13 +107,13 @@ namespace SDDM {
                 d->lastIndex = i;
 
             if (avatarsEnabled) {
-                const QString userFace = QStringLiteral("file://%1/.face.icon").arg(user->homeDir);
-                const QString systemFace = QStringLiteral("file://%1/%2.face.icon").arg(facesDir).arg(user->name);
+                const QString userFace = QStringLiteral("%1/.face.icon").arg(user->homeDir);
+                const QString systemFace = QStringLiteral("%1/%2.face.icon").arg(facesDir).arg(user->name);
 
                 if (QFile::exists(userFace))
-                    user->icon = userFace;
+                    user->icon = QStringLiteral("file://%1").arg(userFace);
                 else if (QFile::exists(systemFace))
-                    user->icon = systemFace;
+                    user->icon = QStringLiteral("file://%1").arg(systemFace);
             }
         }
     }
